### DISCUSSION TITLE
MenuItem grays out after adding Command

### DISCUSSION THREAD

**Discussion by johnabbott72** [AvaloniaUI/Avalonia#12096] [Q&A] [answered]
Having problems with MenuItem from a MenuFlyout - when I add a Command like below it grays out the menu option in preview and runtime - this is in the MainMenuView:
![image](https://github.com/AvaloniaUI/Avalonia/assets/40582159/912c26de-7ab2-4512-9d7c-7a00c17ea080)

Any thoughts as to why?
What I am trying to do is have a home view with navigation and then a main content control that I want to load the apps views into - this is in the HomeView:
![image](https://github.com/AvaloniaUI/Avalonia/assets/40582159/fbeef1cb-856d-44a7-a369-c4227ba577bb)

Any help would be appreciated.

**Comment by sn4k3** [comment]
See: https://github.com/AvaloniaUI/Avalonia/discussions/12062

**Reply by johnabbott72** [reply]
I changed to a method with no parameters but that still didn't work:
![image](https://github.com/AvaloniaUI/Avalonia/assets/40582159/f0cee817-cf60-43e5-976e-fde2d1541b74)

**Reply by sn4k3** [reply]
You can use one parameter but your function must have `object` as parameter type.
If item is still grey out after a parameterless function you should double check your binding (Is it correct?)

**Reply by johnabbott72** [reply, marked as answer]
I did have to adjust the DataContext on the control and then I was able to get it working with a no parameter function.  Thank you for your help.
